src/main/java/edu/wisc/my/restproxy/service/RestProxyServiceImpl.java

```diff
@@ -95,11 +95,15 @@ public Object proxyRequest(final String resourceKey, final HttpServletRequest re
     RequestBody requestBody = null;
     try {
       InputStream inputStream = request.getInputStream();
-      final String contentType = request.getHeader(HttpHeaders.CONTENT_TYPE);
-      if(inputStream != null && MediaType.APPLICATION_JSON_VALUE.equals(contentType)) {
+      final String contentType = request.getContentType();
+      int contentLength = request.getContentLength();
```

Comment:
:+1:  Looks like it should fix the brittleness around content type (e.g. `application/json;charset=utf-8`)